### DISCUSSION TITLE
Apply transformation to the points given in Markups fiducial file

### DISCUSSION THREAD

**Discussion by echonax07** [TorchIO-project/torchio#547] [Q&A] [answered]
Hi 
I am new to medical imaging field.
I have a landmark that specifies different anatomy of the brain in fcsv file(which the 3Dslicer generates). 
These landmarks are ground truth for my model.
So In the data augmentation part, I want to apply the transformation to both training images as well as the landmarks,
Any idea how I could do that?

**Comment by fepegar** [comment, marked as answer]
Hi, @echonax07. We would need some major features added to the library, but I think it's a good idea. It's related to #537, so we can maybe discuss this there. What transforms are you interested in?

**Reply by romainVala** [reply]
I did not realize at first read, but yes that is true if you consider surface as a point cloud, then it is the same datatype as for landmarks. 
Since landmarks are just a list of 3D coordinate, (rigth ?)

there are also a surface format, with triangle (meshes) that is then a little bit different. but may be not for the transform point of vue, because, even with mesh, you end up with a list of 3D coordinate ...

We could then consider the 2 format, mesh and point cloud ... or just only one new torchio datatype is needed

**Reply by fepegar** [reply]
[@romainVala I tried to comment on your issue earlier today but my comment got somehow lost]

I suppose we could add a `Surface` and `Points` classes. It would be nice to make the internal data compatible with ITK transforms, so maybe VTK could be used for this (I think it's not very hard to convert from ITK to VTK nowadays). ITK has classes for mesh and point clouds, but not SimpleITK, unfortunately.

The points class should be able to read ROIs and fiducial files from old (acsv, fcsv) and recent (acsv, .mrk.json) versions of Slicer, and simpler things like .txt or .npy, I suppose.

**Reply by fepegar** [reply]
Here's some handling of point clouds using Python ITK: https://examples.itk.org/src/registration/metricsv4/registertwopointsets/documentation
